compiler/rustc_hir/src/lang_items.rs

```diff
@@ -326,6 +326,8 @@ language_item_table! {
     Range,                   sym::Range,               range_struct,               Target::Struct,         GenericRequirement::None;
     RangeToInclusive,        sym::RangeToInclusive,    range_to_inclusive_struct,  Target::Struct,         GenericRequirement::None;
     RangeTo,                 sym::RangeTo,             range_to_struct,            Target::Struct,         GenericRequirement::None;
+
+    ArraySlice,              sym::array_slice,         array_slice_impl,           Target::Impl,          GenericRequirement::Exact(2);
```

Comment:
Based on this:

https://github.com/rust-lang/rust/blob/76d770ac21d9521db6a92a48c7b3d5b2cc535941/library/core/src/array/mod.rs#L398-L399

I don't think you need this lang item any more.

I suspect you can then remove all the `compiler/` changes in here, as well as the `#[lang = "array_slice"]`.

(It won't work in bootstrap, but you already have a `cfg(not(bootstrap))` on it to deal with that.)

Reply:
Do you mean put `lang = "array"` on the impl block, or get rid of the attribute entirely?

Reply:
I'm hoping you can just get rid of the `#[lang]` attribute entirely on the impl block.

Reply:
Nice, looks like none of the compiler changes were needed now 🎉